### PR TITLE
Add OIDC Logout URL to IDP UI

### PR DESCRIPTION
**Pull request by Chamodii**

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 1]
```diff
@@ -128,6 +128,7 @@
     String clientSecret = null;
     String authzUrl = null;
     String tokenUrl = null;
+    String logoutUrlOidc = null;
```
Comment (Chamodii) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: variable name should be changed to logoutUrlOIDC

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 2]
```diff
@@ -5933,4 +5954,4 @@
     <script type="text/javascript" src="../admin/js/main.js"></script>
     <script type="text/javascript" src="js/idp_mgt_edit.js"></script>
 
-</fmt:bundle>
+</fmt:bundle>
```
Comment (Chamodii) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Add a new line

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 3]
```diff
@@ -1317,15 +1322,16 @@ private static void buildOpenIDConnectAuthenticationConfiguration(IdentityProvid
...
 
         oidcAuthnConfig.setProperties(properties);
         FederatedAuthenticatorConfig[] authenticators = fedIdp.getFederatedAuthenticatorConfigs();
 
         if (paramMap.get("authzUrl") != null && !"".equals(paramMap.get("authzUrl"))
                 && paramMap.get("tokenUrl") != null && !"".equals(paramMap.get("tokenUrl"))
                 && paramMap.get("clientId") != null && !"".equals(paramMap.get("clientId"))
-                && paramMap.get("clientSecret") != null && !"".equals(paramMap.get("clientSecret"))) {
+                && paramMap.get("clientSecret") != null && !"".equals(paramMap.get("clientSecret"))
+                && paramMap.get("logoutUrlOidc") != null && !"".equals(paramMap.get("logoutUrlOidc"))) {
```
Comment (darshanasbg) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Check whether this need to be removed.

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 4]
```diff
@@ -1275,18 +1275,23 @@ private static void buildOpenIDConnectAuthenticationConfiguration(IdentityProvid
         property.setValue(paramMap.get("tokenUrl"));
         properties[2] = property;
 
+        property = new Property();
+        property.setName(IdentityApplicationConstants.Authenticator.OIDC.OIDC_LOGOUT_URL);
+        property.setValue(paramMap.get("logoutUrlOidc"));
```
Comment (darshanasbg) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Check whether "logoutUrlOidc" value is correct as with https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/53

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 5]
```diff
@@ -4698,7 +4707,19 @@
...
+                                           value=<%=Encode.forHtmlAttribute(logoutUrlOIDC)%>>
+
+                                    <div class="sectionHelp">
+                                        <fmt:message key='logout.endpoint.help'/>
+                                    </div>
+                                </td>
+                            </tr>
+                            <tr>
+                                <td class="leftCol-med labelField"><fmt:message key='callbackurl'/>:<span
+                                        class="required">*</span></td>
```
Comment (piraveena) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Any reasons for introducing the "required" field for call backURL?? It was not existing previously

**Comment by piraveena** [comment]
There is an issue with tomcat9 when the number of lines in JSP page exceeds a limit https://github.com/wso2/product-is/issues/6185

**Comment by darshanasbg** [comment]
Resent with: https://github.com/wso2/carbon-identity-framework/pull/2491

**Pull request closed without merging**